rust/src/statistics/formatters/csv_writer.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2025 Sonic Operations Ltd
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at soniclabs.com/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+use std::{fs::File, io::Write, path::Path};
+
+use crate::statistics::{
+    Statistic, StatisticsFormatter,
+    node_count::{NodeCountStatistic, NodeCountsByKindStatistic, NodeCountsByLevelStatistic},
+};
+
+/// A statistics formatter that writes statistics in CSV format.
+pub struct CSVWriter;
+
+impl CSVWriter {
+    /// Initialize a CSV writer for the given statistic suffix.
+    fn init(path: &Path, suffix: &str) -> std::io::Result<File> {
+        File::create(path.join(format!("carmen_stats_{suffix}.csv")))
+    }
+}
+
+/// Writes [`NodeCountsByKindStatistic`] in CSV format.
+fn write_node_counts_by_kind(stat: &NodeCountsByKindStatistic, path: &Path) -> std::io::Result<()> {
+    let mut wtr = CSVWriter::init(path, "node_counts_by_kind")?;
+    wtr.write_all(b"Node Type,Node Subtype,Node Count\n")?;
+    for (type_name, stats) in &stat.aggregated_node_statistics {
+        let node_count = stats.size_count.values().sum::<u64>();
+        wtr.write_all(format!("{type_name},{type_name},{node_count}\n").as_bytes())?;
```

Comment:
```suggestion
```
This can easily be computed in post-processing, removing it makes the format more clear IMO.